src/mbpfan.c

```diff
@@ -211,12 +211,93 @@ t_sensors *retrieve_sensors()
 
 		    s->file = file;
 		    sensors_found++;
+		    if(verbose) {
+		      mbp_log(LOG_INFO, "Found Sensor at %s", path);
+		    }
+
 		}
 
 		free(path);
 		path = NULL;
 	    }
+
 	}
+
+	    // iMac Sensors
+	    // /sys/devices/platform/applesmc.768/
+	    int sensorNo = 0;
+	    char label[80];
+	    char labelType[80] ="";
+	    char labelType2[80] ="";
+	    for(sensorNo = 0; sensorNo<NUM_TEMP_INPUTS; sensorNo++) {
+	      path = smprintf("%s/temp%d_label", APPLESMC_PATH, sensorNo);
+	      FILE *file = fopen(path, "r");
+	      free(path);
+	      if(NULL == file) {
+		continue;
+	      } else {
+		fscanf(file, "%s", label);
+		fclose(file);
+		if (label[1] != 'G' ){ // GPU Only
+		  //		  continue;
+		}
+
+		strncpy(labelType, "???",80);
+		if (label[1] == 'A' ){ strncpy(labelType, "Ambient", 80); }
```

Comment:
Why do you need to copy these?

Reply:
Idea was to later also copy it to s_sensors. So I can later display which sensor triggered the max Temperature.

Reply:
But you can just set `labelType = "Ambient"`?